cmd/router/main_test.go

```diff
@@ -515,3 +536,61 @@ func TestCallServiceWhenMultipleHeadersToPropagate(t *testing.T) {
 	fmt.Printf("final response:%v\n", response)
 	assert.Equal(t, expectedResponse, response)
 }
+
+func TestCallServiceWhenMultipleHeadersToPropagateUsingPatterns(t *testing.T) {
```

Comment:
Let's add a test case where an un-compilable pattern is passed to simulate failure scenario.

Reply:
Added